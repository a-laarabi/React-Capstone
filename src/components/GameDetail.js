import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import { setGames } from '../redux/actions';
import Header from './Header';

const GameDetail = () => {
  const [loading, setLoading] = useState(true);
  const games = useSelector((state) => state.games);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(setGames());
    setLoading(false);
  }, []);

  const { id } = useParams();
  const game = games.find((game) => game.id === +id);
  if (!game) {
    return (
      <div className="spinner-border text-primary" role="status">
        <span className="sr-only">Loading...</span>
      </div>
    );
  }

  return (
    <div className="gameDetail">
      { !loading ? (
        <>

          <div>
            <Header />
            <div className="detailInfo">
              <ul key={game.id}>
                <li>{game.title}</li>
                <li>{game.gener}</li>
                <li>{game.released}</li>
              </ul>
            </div>
            <img src={game.image} alt={game.title} />
            <div className="imgBack" />
          </div>
          <div className="info">
            <ul key={game.id}>
              <li>
                PLATHPHORMS:
                {' '}
                {game.platphorm}
              </li>
              <li>
                GENRE:
                {' '}
                {game.gener}
              </li>
              <li>
                RELEASE DATE:
                {' '}
                {game.released}
              </li>
              <li>
                YEAR START:
                {' '}
                {game.year_start}
              </li>
              <li>
                UPDATED:
                {' '}
                {game.updated}
              </li>
              <li>
                RATING:
                {' '}
                {game.rating}
              </li>
              <li>
                SUGGESTIONS COUNT:
                {' '}
                {game.suggestions_count}
              </li>
              <li>
                PLAYSTIME:
                {' '}
                {game.playtime}
              </li>
              <li>
                STORES:
                {' '}
                {game.stores}
              </li>
              <li>
                TAGS:
                {' '}
                {game.tags}
              </li>
            </ul>
          </div>
        </>

      ) : (
        <div role="status">
          <span className="sr-only">Loading...</span>
        </div>
      )}
    </div>

  );
};

export default GameDetail;
