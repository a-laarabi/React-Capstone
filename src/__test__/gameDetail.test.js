import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import GameDetail from '../components/GameDetail';

describe('Test the components', () => {
  it('Check if the GameDetail.js file\'s section are available', () => {
    const website = renderer.create(
      <Provider store={store}>
        <GameDetail />
      </Provider>,
    );
    expect(website).toMatchSnapshot();
  });
});
