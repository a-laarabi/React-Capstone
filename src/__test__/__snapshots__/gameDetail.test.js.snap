// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the components Check if the GameDetail.js file's section are available 1`] = `
<div
  className="spinner-border text-primary"
  role="status"
>
  <span
    className="sr-only"
  >
    Loading...
  </span>
</div>
`;
